File: domain-layer/src/main/java/com/mohamadamin/cleantvmaze/domain/annotation/ForApplication.java
package com.mohamadamin.cleantvmaze.domain.annotation;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import javax.inject.Qualifier;

/**
 * Explicitly differentiate the object from subviews.
 *
 * @author Saeed Masoumi (saeed@6thsolution.com)
 */
@Qualifier
@Retention(RetentionPolicy.RUNTIME)
public @interface ForApplication {
}

File: domain-layer/src/main/java/com/mohamadamin/cleantvmaze/domain/annotation/ForApplication.kt
package com.mohamadamin.cleantvmaze.domain.annotation

import javax.inject.Qualifier

/**
 * Explicitly differentiate the object from subviews.
 * @author MohamadAmin Mohamadi (mohammadi.mohamadamin@gmail.com) on 5/6/17.
 */

@Qualifier
@Retention(AnnotationRetention.RUNTIME)
annotation class ForApplication